Framework/API/src/ExperimentInfo.cpp

```diff
@@ -1004,7 +1004,8 @@ ExperimentInfo::getInstrumentFilename(const std::string &instrumentName,
     // find the first beat file
     for (Poco::DirectoryIterator dir_itr(directoryName); dir_itr != end_iter;
          ++dir_itr) {
-      if (!Poco::File(dir_itr->path()).isFile())
+
+      if (!Poco::Path(dir_itr->path()).isFile())
```

Comment:
Is this performing an unnecessary copy?

`dir_itr->path()` returns a `Poco::Path` object anyway so we could just do `if (!dir_itr->path().isFile())` ?